nexus/src/app/update.rs

```diff
@@ -149,4 +157,170 @@ impl super::Nexus {
             .await
             .map_err(HttpError::from)
     }
+
+    /// Get external update status with aggregated component counts
+    pub async fn update_status_external(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<views::UpdateStatus, Error> {
+        // ? because we expect there to always be a current target release. but
+        // it can still have an Unspecified release_source
+        let db_target_release =
+            self.datastore().target_release_get_current(opctx).await?;
+        let target_release = match db_target_release.tuf_repo_id {
+            Some(tuf_repo_id) => {
+                let version = self
+                    .datastore()
+                    .tuf_repo_get_version(opctx, &tuf_repo_id)
+                    .await?;
+                Some(views::TargetRelease {
+                    time_requested: db_target_release.time_requested,
+                    version,
+                })
+            }
+            None => None,
+        };
+
+        let components_by_release_version =
+            self.component_version_counts(opctx, &db_target_release).await?;
+
+        let time_last_progress = self
+            .datastore()
+            .blueprint_target_get_current(opctx)
+            .await?
+            .time_made_target;
+
+        Ok(views::UpdateStatus {
+            target_release: Nullable(target_release),
+            components_by_release_version,
+            time_last_progress,
+        })
+    }
+
+    /// Get component status using read-only queries to avoid batch operations
+    async fn component_version_counts(
+        &self,
+        opctx: &OpContext,
+        target_release: &nexus_db_model::TargetRelease,
+    ) -> Result<BTreeMap<String, usize>, Error> {
+        // Get the latest inventory collection
+        let Some(inventory) =
+            self.datastore().inventory_get_latest_collection(opctx).await?
+        else {
+            // No inventory collection available, return empty counts
+            return Ok(BTreeMap::new());
+        };
+
+        // Build current TargetReleaseDescription, defaulting to Initial if
+        // there is no tuf repo ID which, based on DB constraints, happens if
+        // and only if target_release_source is 'unspecified', which should only
+        // happen in the initial state before any target release has been set
+        let curr_target_desc = match target_release.tuf_repo_id {
+            Some(id) => TargetReleaseDescription::TufRepo(
+                self.datastore()
+                    .tuf_repo_get_by_id(opctx, id.into())
```

Comment:
I guess we wind up doing this query twice in `update_status_external()`?  The other one uses `tuf_repo_get_version()`, which the comment says is the same as this one but avoids fetching artifact information.  If we're getting it anyway, maybe we could use this where we needed it earlier.  Obviously would involve some refactoring.